ui/src/views/AutogenView.vue

```diff
@@ -989,7 +989,7 @@ export default {
               }
               break
             }
-            if (!input) {
+            if (!input && input !== 0) {
```

Comment:
should this validation be moved to line 982 along with the existing validation? moving it there might simplify all these if conditions and streamline the iteration

Reply:
@fdalsotto that may affect code at lines 985-991. Do you have a suggestion to restructure whole bit?

Reply:
I was going to say that line 982 should look like:

if (param.name !== key || (!input && input !== 0))) {

though this will impact line 985. 

the logic overall looks a bit confusing though. Maybe a rewrite would make sense here.